### PR TITLE
Fixed ancient pedestal bug #499

### PR DESCRIPTION
**Pull request by Redemption198**
I've added a check in the Pedestal listener, now if the block above the pedestal is not air, the item will not be added and the player will be notified about that. 

P.S. The item can still be pushed by a piston, I don't know how this can be fixed.

### CODE REVIEW

**Comment by Redemption198** [comment]
@xH3LLRAIZ3Rx reported this issue on (https://github.com/TheBusyBiscuit/Slimefun4/issues/499)

File: src/me/mrCookieSlime/Slimefun/Setup/Messages.java [hunk 1]
```diff
@@ -86,6 +86,7 @@ public static void setup() {
 		local.setDefault("machines.ANCIENT_ALTAR.unknown-catalyst", "&4Unknown Catalyst! &cUse the correct Recipe instead!");
 		local.setDefault("machines.ANCIENT_ALTAR.unknown-recipe", "&4Unknown Recipe! &cUse the correct Recipe instead!");
 		local.setDefault("machines.HOLOGRAM_PROJECTOR.enter-text", "&7Please enter your desired Hologram Text in your Chat. &r(Color Codes are supported!)");
+		local.setDefault("PEDESTAL.obstructed", "&4Pedestal is obstructed! &cRemove the block above the pedestal!");
```
Comment (Poslovitch) [inline src/me/mrCookieSlime/Slimefun/Setup/Messages.java]: Please use machines.ANCIENT_PEDESTAL.obstructed instead ;)

Comment (Redemption198) [reply src/me/mrCookieSlime/Slimefun/Setup/Messages.java]: Done :)

**Comment by TheBusyBiscuit** [comment]
This is an attempt to 'avoid' the Issue, but doesn't fix it.
What if you place an Item, then a Block?

That is also what this Issue was actually about.
This is halfly done, you could prevent the block placement as well.

**Comment by Redemption198** [comment]
I've added the event that prevents the block placement, but water and pistons can still push away the item.

**Comment by Redemption198** [comment]
@Poslovitch without line 188 the server throws org.bukkit.event.EventException: null

**Comment by Poslovitch** [comment]
Okay gotcha

File: src/me/mrCookieSlime/Slimefun/Setup/Messages.java [hunk 2]
```diff
@@ -72,6 +72,7 @@ public static void setup() {
 		local.setDefault("commands.research.reset-target", "&cYour Knowledge has been reset");
 		local.setDefault("machines.in-use", "&cThis Block's Inventory is currently opened by a different Player.");
 		local.setDefault("machines.ignition-chamber-no-flint", "&cIgnition Chamber is missing Flint and Steel.");
+    		local.setDefault("messages.cant-place" ,"&cYou can't place that block there!");
```
Comment (Poslovitch) [inline src/me/mrCookieSlime/Slimefun/Setup/Messages.java]: Change "messages.cant-place" to "messages.cannot-place". This is a detail, but we have to keep a "neutral" english for convenience reasons.

**Comment by Poslovitch** [comment]
Then, this will be good to go for me :)

**Comment by Redemption198** [comment]
@Poslovitch thanks for your advice :)

File: src/me/mrCookieSlime/Slimefun/listeners/AncientAltarListener.java [hunk 3]
```diff
@@ -174,5 +179,17 @@ else if (!e.getItem().hasMetadata("no_pickup") && e.getItem().getItemStack().has
 			e.getItem().remove();
 		}
 	}
+	
+	@EventHandler
+	public void onBlockPlace(BlockPlaceEvent e) {
+		if(e.isCancelled()) return;
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/listeners/AncientAltarListener.java]: Use @EventHandler(ignoreCancelled=true) instead.

Comment (TheBusyBiscuit) [reply src/me/mrCookieSlime/Slimefun/listeners/AncientAltarListener.java]: Also you should give it a priority.

**Pull request merged**